### PR TITLE
Fix IMAPBodyPart.getFileName()

### PR DESCRIPTION
**Pull request by fred84**
Fix for #330 

P.S. I signed OCA with email sergey@galkin.me

### CODE REVIEW

**Comment by fred84** [comment]
@bshannon Test added.

File: mail/src/main/java/com/sun/mail/imap/IMAPBodyPart.java [hunk 1]
```diff
@@ -168,7 +168,7 @@ public String getFileName() throws MessagingException {
 	String filename = null;
 	if (bs.dParams != null)
 	    filename = bs.dParams.get("filename");
-	if (filename == null && bs.cParams != null)
+	if ((filename == null || filename.equals("")) && bs.cParams != null)
```
Comment (tweimer) [inline mail/src/main/java/com/sun/mail/imap/IMAPBodyPart.java]: How about using filename.isEmpty() here?

Comment (fred84) [reply mail/src/main/java/com/sun/mail/imap/IMAPBodyPart.java]: agree

**Comment by bshannon** [comment]
This looks good.  Can you please file an issue that describes the problem, including a
sample message that exhibited this problem, and reference that issue from this pull request?

Thanks.

**Comment by fred84** [comment]
@bshannon 
- issue created: #330 
- PR description updated

**Comment by bshannon** [comment]
Your OCA was approved so this is almost ready to integrate.  One more small thing...

Can you add an entry to doc/CHANGES.txt listing the bug that this will fix?

Once that's done, I'll merge this pull request.

Thanks!

**Comment by fred84** [comment]
@bshannon 

CHANGES.txt updated.

**Pull request merged**